tests/unit/wordpress_mock.py

```diff
@@ -0,0 +1,526 @@
+        """Simulate activate a WordPress plugin.
+
+        Args:
+            plugin: plugin name.
+        """
+        self.activated_plugins.add(plugin)
+
+    def deactivate_plugin(self, plugin: str) -> None:
+        """Simulate deactivate a WordPress plugin.
+
+        Args:
+            plugin: plugin name.
+        """
+        self.activated_plugins.remove(plugin)
+
+    def activate_theme(self, theme: str) -> None:
+        """Simulate activate a WordPress theme.
+
+        Args:
+            theme: theme name.
+        """
+        self.activated_theme = theme
+
+    def update_option(self, name: str, value: str | dict) -> None:
+        """Simulate update a WordPress option.
+
+        Args:
+            name: option name.
+            value: option value, which can be a string for PHP string or a dict for PHP array.
+        """
+        self.options[name] = value
+
+    def delete_option(self, name: str) -> None:
+        """Simulate delete a WordPress option.
+
+        Args:
+            name: option name.
+        """
+        try:
+            del self.options[name]
+        except KeyError:
+            pass
+
+
+class WordpressDatabaseMock:
+    """Simulate database interaction like connecting, login, WordPress installation."""
+
+    def __init__(
+        self,
+        builtin_wordpress_options: typing.Optional[typing.Dict[str, typing.Dict | str]] = None,
+    ) -> None:
+        """Initialize the instance.
+
+        Args:
+            builtin_wordpress_options: some builtin WordPress options come with the
+                WordPress installation.
+        """
+        self._databases = {}
+        self._database_credentials = {}
+        self._builtin_wordpress_options = builtin_wordpress_options
+
+    @staticmethod
+    def _database_identifier(host: str, database: str):
+        """Create a key for index simulated databases.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        return host, database
+
+    def prepare_database(self, host: str, database: str, user: str, password: str) -> None:
+        """Set up a simulated database, so it can be connected and installed with WordPress.
+
+        Args:
+            host: database host.
+            database: database name.
+            user: database user.
+            password: database password.
+        """
+        key = self._database_identifier(host, database)
+        if key in self._databases:
+            raise KeyError(f"Database (host={host}, database={database}) already exists")
+        self._databases[key] = None
+        self._database_credentials[key] = {"user": user, "password": password}
+
+    def database_can_connect(self, host: str, database: str) -> bool:
+        """Test if given host and database can connect to a simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
```

Comment:
Should raise an error in case the key does not exist? Would be worth it to document the return in docstrings?

Reply:
Fixed, thanks!